streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -1780,13 +1781,24 @@ public Map<String, Object> getGlobalConsumerConfigs(final String clientId) {
      * @return Map of the producer configuration.
      */
     @SuppressWarnings("WeakerAccess")
-    public Map<String, Object> getProducerConfigs(final String clientId) {
+    public Map<String, Object> getProducerConfigs(final String clientId,
+                                                  final UUID processId,
+                                                  final int threadIdx) {
         final Map<String, Object> clientProvidedProps = getClientPropsWithPrefix(PRODUCER_PREFIX, ProducerConfig.configNames());
 
         checkIfUnexpectedUserSpecifiedConsumerConfig(clientProvidedProps, NON_CONFIGURABLE_PRODUCER_EOS_CONFIGS);
 
         // generate producer configs from original properties and overridden maps
-        final Map<String, Object> props = new HashMap<>(eosEnabled ? PRODUCER_EOS_OVERRIDES : PRODUCER_DEFAULT_OVERRIDES);
+        final Map<String, Object> props;
+        if (eosEnabled) {
+            props = new HashMap<>(PRODUCER_EOS_OVERRIDES);
+            props.put(
+                ProducerConfig.TRANSACTIONAL_ID_CONFIG,
+                getString(StreamsConfig.APPLICATION_ID_CONFIG) + "-" + processId + "-" + threadIdx
```

Comment:
Given that we don't create a producer per task any longer, we can set the tx-id in a centralized place.